kernel/src/main/java/org/kframework/parser/KRead.java

```diff
@@ -45,6 +59,14 @@ public K prettyRead(Module mod, Sort sort, CompiledDefinition def, Source source
         }
     }
 
+    private K parseKore(Module mod) {
+        try {
+            return  KoreToK.parseKoreToK(options.fileToParse(), idsToLabelsProvider.getKoreToKLabels(), mod.sortAttributesFor());
+        } catch (ParseError parseError) {
+            throw KEMException.criticalError("Parse error" );
+        }
+    }
```

Comment:
I don't like that `KRead` has a `KastOptions` put into it. Can we instead make this function just take a string to parse, and assume that reading the file has been handled elsewhere?

The main reason is that we intend to make KRun and KProve go through KRead as well (so that we can avoid parse-time by passing in JSON input or BINARY input, for example), and they do their own file-handling code.

Reply:
Note also that @gtrepta is working on making KRun go through KRead right now, which will make a new `KReadOptions.java` available. Currently we're only giving it the `--input` options, and we'll include `KReadOptions` into `KRunOptions` and `KastOptions`. But if there are more parameters which are inherent to KRead, we can move those over too once he's done.

But I'd like to do the refactoring incrementally to reduce chances of bugs, so we're just focusing on getting KRead to be used directly by other parts of the code (and the other parts of the code need to do their own file handling), then we'll refactor it to localize more of the file handling code into KRead directly.

Reply:
We could, but the Kore parsing code is in Scala and doesn't have a parse from string option and I was reluctant to add it. On the other hand, isn't it less efficient to pass a string, esp. Considering that the input can sometimes be quite large?